runelite-client/src/main/java/net/runelite/client/plugins/microbot/bankjs/BanksShopper/BanksShopperPlugin.java

```diff
@@ -63,15 +90,53 @@ public void onChatMessage(ChatMessage event) {
                 Microbot.status = "[Shutting down] - Reason: Not enough coins.";
                 Microbot.showMessage(Microbot.status);
                 banksShopperScript.shutdown();
-                Rs2Shop.closeShop();
-                if (config.logout()) {
-                    Microbot.getClientThread().runOnSeperateThread(() -> {
-                        sleep(1200);
-                        Rs2Player.logout();
-                        return true;
-                    });
-                }
             }
         }
     }
+    
+    @Subscribe
+    public void onConfigChanged(ConfigChanged event) {
+        if (!event.getGroup().equals(BanksShopperConfig.configGroup)) return;
+        
+        if (event.getKey().equals(BanksShopperConfig.npcName)) {
+            npcName = config.npcName();
```

Comment:
I think it would make more sense to call BankShopperScript.npcName instead of defining the npcName in the plugin and injecting the plugin inside the script which is an anti pattern because plugin contains an instance of the BankShopperScript, which creates a infinite loop of references.

Reply:
I dont like using static fields for the plugin. I have fixed the aforementioned issue by making the script private inside of the plugin and removing the @Inject annotation from the constructor of the Script class.

You can no longer call plugin.script.plugin.script. etc etc.